contentcuration/automation/utils/appnexus/base.py

```diff
@@ -1,46 +1,183 @@
+import time
+import logging
+import requests
 from abc import ABC
 from abc import abstractmethod
 from builtins import NotImplementedError
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from . import errors
+
+
+class SessionWithMaxConnectionAge(requests.Session):
+    """
+        Session with a maximum connection age. If the connection is older than the specified age, it will be closed and a new one will be created.
+        The age is specified in seconds.
+    """
+    def __init__(self, age = 10):
+        super().__init__()
+        self.age = age
+        self.last_used = time.time()
+
+    def request(self, *args, **kwargs):
+        current_time = time.time()
+        if current_time - self.last_used > self.age:
+            self.close()
+
+        self.last_used = current_time
+
+        return super().request(*args, **kwargs)
 
 
 class BackendRequest(object):
-    """ Class that should be inherited by specific backend for its requests"""
-    pass
+    """ Class that holds the request information for the backend """
+    def __init__(
+        self,
+        method,
+        path,
+        params=None,
+        data=None,
+        json=None,
+        headers=None,
+        timeout=(5, 10),
```

Comment:
10 seconds is too short for a request timeout

Reply:
Wooooopps I misread the number of seconds in your comment.

Reply:
Thank you @bjester! I updated the value :)